app/src/main/java/org/wikipedia/analytics/metricsplatform/MetricsEvent.kt

```diff
@@ -70,22 +93,54 @@ open class MetricsEvent {
             pageTitle.namespace().code(),
             Namespace.of(pageTitle.namespace().code()).toString(),
             revisionId,
-            "",
-            pageTitle.wikiSite.languageCode,
-            null, null, null)
+            null,
+            pageTitle.wikiSite.languageCode
+        )
     }
 
     private fun getPerformerData(): PerformerData {
         return PerformerData(
+            AccountUtil.hashCode(),
             AccountUtil.userName,
             AccountUtil.isLoggedIn,
-            AccountUtil.hashCode(),
+            null,
             EventPlatformClient.AssociationController.sessionId,
             EventPlatformClient.AssociationController.pageViewId,
             AccountUtil.groups,
-            null,
-            WikipediaApp.instance.languageState.appLanguageCode,
             WikipediaApp.instance.languageState.appLanguageCodes.toString(),
+            WikipediaApp.instance.languageState.appLanguageCode,
+            null
+        )
+    }
+
+    protected fun getInteractionData(
+        action: String,
+        actionSubtype: String?,
+        actionSource: String?,
+        actionContext: String?,
+        elementId: String?,
+        elementFriendlyName: String?,
+        funnelEntryToken: String?,
+        funnelEventSequencePosition: Int?
```

Comment:
For these nullable parameters, you can give them default values like:
```
actionSubtype: String? = null
```

And then In line 139, you can just call `return InteractionData(action)` without sending `null` for the other parameters.

Reply:
thanks @cooltey -- i guess i can get rid of `getInteractionDataMinimal()` altogether in L139 -- thanks!